pkg/hub/controllers/slice_controller.go

```diff
@@ -408,25 +419,117 @@ func (r *SliceReconciler) getComponentStatus(ctx context.Context, c *component,
-		return nil, nil
-	}
 	if len(pods) == 0 {
+		if c.ignoreMissing {
+			log.Info("ignore missing pod for ", "component", c.name)
+			return nil, nil
+		}
 		log.Error(fmt.Errorf("no pods running"), "unhealthy", "pod", c.name)
 		cs.ComponentHealthStatus = spokev1alpha1.ComponentHealthStatusError
 		return cs, nil
 	}
+	// TODO: verify "PodConditionType == ContainersReady" when
+	// readiness-probe for kubeslice components are implemented
 	for _, pod := range pods {
 		if pod.Status.Phase != corev1.PodRunning {
-			log.Info("pod is not healthy", "component", c.name)
+			log.Info("pod is not in running phase", "component", c.name)
 			cs.ComponentHealthStatus = spokev1alpha1.ComponentHealthStatusError
 			return cs, nil
+		} else {
+			for _, containerStatus := range pod.Status.ContainerStatuses {
+				terminatedState := containerStatus.State.Terminated
+				if terminatedState != nil && terminatedState.ExitCode != 0 {
+					log.Info("container terminated with non-zero exitcode",
+						"component", c.name,
+						"container", containerStatus.Name,
+						"exitcode", terminatedState.ExitCode)
+					cs.ComponentHealthStatus = spokev1alpha1.ComponentHealthStatusError
+					break
+				}
+			}
 		}
 	}
+	log.Info("health status normal", "component", c.name)
 	cs.ComponentHealthStatus = spokev1alpha1.ComponentHealthStatusNormal
 	return cs, nil
 }
 
+func (r *SliceReconciler) fetchTunnelStatus(ctx context.Context, c *component) (*spokev1alpha1.ComponentStatus, error) {
+	log := logger.FromContext(ctx)
+	log.Info("health check for gateway tunnel")
+	tunnel := &spokev1alpha1.ComponentStatus{
+		Component: c.name,
+	}
+	sliceGwDeployments := &appsv1.DeploymentList{}
+	listOpts := []client.ListOption{
+		client.MatchingLabels(c.labels),
+		client.InNamespace(c.ns),
+	}
+	err := r.MeshClient.List(ctx, sliceGwDeployments, listOpts...)
+	if err != nil {
+		log.Error(err, "Could not list the slicegw deployments", "component", c.name)
+		return nil, err
+	}
+	//1. zero number of deployments -> ignore the slicegw tunnel status
+	if len(sliceGwDeployments.Items) == 0 {
+		log.Info("SliceGW deployments are not present, skipping health check", "component", c.name)
+		return nil, nil
+	}
+	//2. non zero number of deployments for slicegw -> fetch status of all pods
+	podList := &corev1.PodList{}
+	if err := r.MeshClient.List(ctx, podList, listOpts...); err != nil {
+		log.Error(err, "Failed to list pods", "component", c.name)
+		tunnel.ComponentHealthStatus = spokev1alpha1.ComponentHealthStatusError
+		return tunnel, nil
+	}
+	pods := podList.Items
+	if len(pods) == 0 {
+		log.Error(fmt.Errorf("no gw pods running"), "error state", "component", c.name)
+		tunnel.ComponentHealthStatus = spokev1alpha1.ComponentHealthStatusError
+		return tunnel, nil
+	}
+	tunStatus_sum := 0     // status sum zero implies all gateways are healthy
+	tunStatus_product := 1 // status product zero implies atleast one gw is healthy
+	if len(pods) != len(sliceGwDeployments.Items) {
```

Comment:
This will fail if gw redundancy is enabled. Every deployment will have two pods so this conditional will fail always.

Reply:
I think that this check needs to be based on number of attached clusters. But since according to the current implementation, when gw redundancy is enabled, two separate gateway deployments are created. Each manages only one gw pod! Thus this works But It would be better If we could revisit this.